frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -156,159 +140,139 @@ class DashboardInner extends Component {
   };
 
   onAddQuestion = () => {
-    const { dashboard } = this.props;
-    this.setEditing(dashboard);
+    this.setEditing(this.props.dashboard);
     this.props.toggleSidebar(SIDEBAR_NAME.addQuestion);
   };
 
-  renderContent = () => {
-    const { dashboard, selectedTabId, isNightMode, isFullscreen } = this.props;
-
-    const canWrite = dashboard?.can_write ?? false;
-
-    const dashboardHasCards = dashboard?.dashcards.length > 0 ?? false;
+  shouldRenderAsNightMode() {
+    return this.props.isNightMode && this.props.isFullscreen;
+  }
 
+  renderContent = () => {
+    const p = this.props;
+    const canWrite = p.dashboard?.can_write ?? false;
+    const dashboardHasCards = p.dashboard?.dashcards.length > 0 ?? false;
     const tabHasCards =
-      dashboard?.dashcards.filter(
+      p.dashboard?.dashcards.filter(
         c =>
-          selectedTabId !== undefined && c.dashboard_tab_id === selectedTabId,
+          p.selectedTabId !== undefined &&
+          c.dashboard_tab_id === p.selectedTabId,
       ).length > 0 ?? false;
 
-    const shouldRenderAsNightMode = isNightMode && isFullscreen;
-
     if (!dashboardHasCards && !canWrite) {
       return (
         <DashboardEmptyStateWithoutAddPrompt
-          isNightMode={shouldRenderAsNightMode}
+          isNightMode={this.shouldRenderAsNightMode()}
         />
       );
     }
+
     if (!dashboardHasCards) {
       return (
         <DashboardEmptyState
-          dashboard={dashboard}
-          isNightMode={shouldRenderAsNightMode}
+          dashboard={p.dashboard}
+          isNightMode={this.shouldRenderAsNightMode()}
           addQuestion={this.onAddQuestion}
-          closeNavbar={this.props.closeNavbar}
+          closeNavbar={p.closeNavbar}
         />
       );
     }
+
     if (dashboardHasCards && !tabHasCards) {
       return (
         <DashboardEmptyStateWithoutAddPrompt
-          isNightMode={shouldRenderAsNightMode}
+          isNightMode={this.shouldRenderAsNightMode()}
         />
       );
     }
+
     return (
       <DashboardGridConnected
         {...this.props}
-        dashboard={this.props.dashboard}
-        isNightMode={shouldRenderAsNightMode}
+        dashboard={p.dashboard}
+        isNightMode={this.shouldRenderAsNightMode()}
         onEditingChange={this.setEditing}
       />
     );
   };
 
   render() {
-    const {
-      addParameter,
-      dashboard,
-      isEditing,
-      isEditingParameter,
-      isFullscreen,
-      isNightMode,
-      isSharing,
-      parameters,
-      parameterValues,
-      draftParameterValues,
-      editingParameter,
-      setParameterValue,
-      setParameterIndex,
-      setEditingParameter,
-      isHeaderVisible,
-      isAutoApplyFilters,
-    } = this.props;
-
-    const { error, isParametersWidgetSticky } = this.state;
-
-    const shouldRenderAsNightMode = isNightMode && isFullscreen;
-
-    const visibleParameters = getVisibleParameters(parameters);
+    const p = this.props;
```

Comment:
Same here: instead of wasting lines on avoiding writing `this.props`, I am putting it into a one-char variable that's almost invisible to the eye but _and_ removes this unnecessary cruft.